File: python/bob/bob.py
import re 

class Bob:
  def hey(self, input):
    if input.rstrip() == "":
      return "Fine. Be that way!"
    elif input == input.upper() and re.search('[A-Z]', input):
      return "Woah, chill out!"
    elif input.rstrip()[-1] == "?":
      return "Sure."
    else:
      return "Whatever." 

File: clojure/bob/bob.clj
(ns bob)

(defn is-scream [user-input]
  (if (and (= user-input (.toUpperCase user-input)) 
    (not= user-input (.toLowerCase user-input))) 
      true
      false))

(defn is-question [user-input]
  (.endsWith user-input "?"))

(defn response-for [user-input]
  (cond
    (empty? (.trim user-input))  "Fine. Be that way!"
    (is-scream (.trim user-input))"Woah, chill out!"
    (is-question (.trim user-input))"Sure."
    :else "Whatever."))